VA-4212 Publish to Social Jobs

howardrigberg opened the pull request:
#### Issue
- https://vimean.atlassian.net/browse/VA-4212
- https://vimean.atlassian.net/browse/VA-4213

#### Summary
Added support for GET and PUT requests to /videos/{video_id}/publish_to_social

#### How to Test
Test setup is a bit complicated. See me IRL if you want instructions for testing in the app.

howardrigberg committed:
adding support for getting a publish job

howardrigberg committed:
Adding support for a PUT request to publish_to_social

howardrigberg committed:
changing page id data type

howardrigberg committed:
changing post field naming

howardrigberg committed:
changing category id to a string

howardrigberg committed:
some clean up

howardrigberg committed:
modifying docs

anthonycr commented on models/src/main/java/com/vimeo/networking2/PublishToFacebookPost.kt:
```diff
@@ -0,0 +1,61 @@
+         * The title of the post as it will appear on Facebook.
+         */
+        @Json(name = "title")
+        val title: String,
```
This and all properties in this class should be nullable with a default value of null. This comment also applies to
- `PublishToLinkedInPost`
- `PublishToTwitterPost`
- `PublishToYouTubePost`

Unless the API ignores field filters for these non null properties they should be made null.

anthonycr commented on models/src/main/java/com/vimeo/networking2/PublishToYouTubePost.kt:
```diff
@@ -0,0 +1,44 @@
+         * private: The video will not be publicly available, and only viewable to the owner.
+         */
+        @Json(name = "privacy")
+        val privacy: String,
```
Should have a corresponding enum if I understand the documentation correclty

WhosNickDoglio reviewed (approved):
The only change I was going to mention was the enum for Privacy that @anthonycr already mentioned. 

Everything else looks good. 👍

howardrigberg committed:
adding params package and using enum for youtube privacy

howardrigberg committed:
removing annotation for enum

howardrigberg committed:
trying different annotation

howardrigberg committed:
adding import

howardrigberg committed:
using gson annotations

howardrigberg committed:
using gson annotations for all parameters

howardrigberg committed:
adding explicit cache header to putPublishJob

anthonycr commented:
@howardrigberg I can review after you resolve the merge conflicts.

howardrigberg committed:
testing full serializable on post parameters

howardrigberg committed:
Revert "testing full serializable on post parameters"

This reverts commit 1d7b2e57d3e35ad21e7fed06b9d94b74d9dff4b2.

howardrigberg committed:
Merge branch 'develop' into VA-4212-Publish-To-Social

howardrigberg committed:
merging with develop and resolving confilcts

changing PublishJobDestinations to PublishJobAttempts in the context of Video connections as it more accurately reflects the data and resolves  naming conflcts

howardrigberg commented:
@anthonycr  Thanks for the review. I moved the param objects into a new package called "params" to distinguish them from the other models. I added the necessary enum and used  `@SerializedName` to ensure that all params will get serialized to the body of the put request correctly.

@WhosNickDoglio I renamed your `PublishJobDestinations` class to `PublishJobAttempts` to resolve a merge conflict and give the class a name that more accurately reflects its purpose. Please let me know if it passes your functional tests and that you agree with my changes.

WhosNickDoglio commented:
> @WhosNickDoglio I renamed your `PublishJobDestinations` class to `PublishJobAttempts` to resolve a merge conflict and give the class a name that more accurately reflects its purpose. Please let me know if it passes your functional tests and that you agree with my changes.

@howardrigberg  No issues with name changes and just tested and everything looks good. 👍

anthonycr commented on models/build.gradle:
```diff
@@ -19,6 +19,8 @@ dependencies {
     kapt "com.squareup.moshi:moshi-kotlin-codegen:1.8.0"
     implementation "com.squareup.moshi:moshi:1.8.0"
 
+    api "com.squareup.retrofit2:converter-gson:2.5.0"
```
Please use Moshi, we are not using Gson going forward.

howardrigberg committed:
trying to avoid gson in models

howardrigberg committed:
removing gson dependency from gradle

anthonycr reviewed (approved):
LGTM!
